src/users/users.service.ts

```diff
@@ -73,4 +73,48 @@ export class UsersService {
       })
       .exec();
   }
+
+  async updateRole(id: Types.ObjectId, updateRole: UpdateRoleDto) {
+    await this.userModel
+      .updateOne(
+        { _id: id },
+        {
+          role: updateRole.role,
+        },
+      )
+      .exec();
+  }
+
+  async updateUserInfo(userId: Types.ObjectId, updateInfo: UpdateInfoDto) {
+    const user = await this.userModel
+      .findOne({ _id: userId }, { password: true })
+      .exec();
+    if (await argon2.verify(user.password, updateInfo.oldPassword)) {
+      if (updateInfo.newUsername) {
+        await this.userModel
+          .updateOne(
+            { _id: userId },
+            {
+              username: updateInfo.newUsername,
+            },
+          )
+          .exec();
+      }
+      if (updateInfo.newPassword) {
+        const hashedPassword = await argon2.hash(updateInfo.newPassword);
+        await this.userModel
+          .updateOne(
+            { _id: userId },
+            {
+              password: hashedPassword,
+            },
+          )
+          .exec();
+      }
```

Comment:
So there will be 2 queries if I wanna update both username and password right?
Can it both be updated in one query?